src/handlers/token_handler.erl

```diff
@@ -0,0 +1,32 @@
+-module(token_handler).
+
+-export([init/3,
+         allowed_methods/2,
+         content_types_provided/2,
+         is_authorized/2,
+         to_json/2]).
+
+init(_Transport, _Req, []) ->
+    {upgrade, protocol, cowboy_rest}.
+
+allowed_methods(Req, State) ->
+    {[<<"GET">>], Req, State}.
+
+is_authorized(Req, State) ->
+    case cowboy_req:parse_header(<<"authorization">>, Req) of
+        {ok, {<<"basic">>, {Email, Password}}, Req2} ->
+            case db_user:authenticate(Email, Password) of
+                {ok, User} ->
+                    {true, Req2, #{user => User}};
+                _ -> {{false, <<"Basic realm=\"holidayping\"">>}, Req2, State}
```

Comment:
If you are using basic auth, you will probably use it in many places. Move this check to an utility function. If many handlers use repeated functions, use https://github.com/chef/mixer to have Ruby like mixins in Erlang. It is not as good as Clojure/Elixir protocols, but it is really useful.